File: src/test/java/im/tox/tox4j/DhtNode.java
package im.tox.tox4j;

public class DhtNode {
    public final String ipv4;
    public final String ipv6;
    public final int tcpPort;
    public final int udpPort;
    public final byte[] dhtId;

    public DhtNode(String ipv4, String ipv6, int port, String dhtId) {
        this.ipv4 = ipv4;
        this.ipv6 = ipv6;
        this.tcpPort = port;
        this.udpPort = port;
        this.dhtId = ToxCoreTestBase.parsePublicKey(dhtId);
    }

    public DhtNode(String ipv4, String ipv6, int udpPort, int tcpPort, String dhtId) {
        this.ipv4 = ipv4;
        this.ipv6 = ipv6;
        this.tcpPort = tcpPort;
        this.udpPort = udpPort;
        this.dhtId = ToxCoreTestBase.parsePublicKey(dhtId);
    }

}

File: src/test/java/im/tox/tox4j/DhtNode.scala
package im.tox.tox4j

final case class DhtNode(ipv4: String, ipv6: String, udpPort: Int, tcpPort: Int, dhtId: Array[Byte]) {

  def this(ipv4: String, ipv6: String, udpPort: Int, tcpPort: Int, dhtId: String) {
    this(ipv4, ipv6, udpPort, tcpPort, ToxCoreTestBase.parsePublicKey(dhtId))
  }

  def this(ipv4: String, ipv6: String, port: Int, dhtId: String) {
    this(ipv4, ipv6, port, port, dhtId)
  }

}
